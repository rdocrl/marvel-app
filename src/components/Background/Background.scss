.background {
  flex: 1;
  background-size: cover;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  &__opacity {
    position: absolute;
    background: #000;
    height: 100%;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    opacity: 98%;
  }

  > div:not(.background__opacity) {
    position: relative;
  }
}

.content {
  position: relative;
  z-index: 1;
}
