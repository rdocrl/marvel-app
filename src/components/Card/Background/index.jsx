import './Background.scss';

const Background = ({ image, children }) => {
  return (
    <>
      <div
        className="background"
        style={{
          backgroundImage: `url('${image}')`
        }}>
        <div className="background__opacity"></div>
      </div>
      <div className="content">{children}</div>
    </>
  );
};

export default Background;
