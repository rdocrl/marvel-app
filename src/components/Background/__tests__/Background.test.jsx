import { render } from '@testing-library/react';
import Background from '../';

describe('<Background /> tests', () => {
  describe('Snapshots', () => {
    it('Should render', () => {
      const { asFragment } = render(<Background />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
