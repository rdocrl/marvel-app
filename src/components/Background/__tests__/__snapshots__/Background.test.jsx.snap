// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> tests Snapshots Should render 1`] = `
<DocumentFragment>
  <div
    class="background"
    style="background-image: url(undefined);"
  >
    <div
      class="background__opacity"
    />
  </div>
  <div
    class="content"
  />
</DocumentFragment>
`;
